File: medplum-java/medplum-model/src/main/java/com/fhir/SpecimenDefinition_Additive.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSpecimenDefinition_Additive.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SpecimenDefinition_Additive {
  Optional<String> id();

  Optional<List<Extension>> extension();

  Optional<List<Extension>> modifierExtension();

  Optional<CodeableConcept> additiveCodeableConcept();

  Optional<Reference> additiveReference();

  static ImmutableSpecimenDefinition_Additive.Builder builder() {
    return ImmutableSpecimenDefinition_Additive.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/SpecimenDefinition_Additive.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Reference } from "./Reference";

export interface SpecimenDefinition_Additive {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    additiveCodeableConcept: CodeableConcept | null | undefined;
    additiveReference: Reference | null | undefined;
}
